src/Routing.jl

```diff
@@ -204,44 +201,61 @@ function update_state_from_channel!(
             x = dist[1]
             distribution = [
                 x / (2 * t * sqrt(π * D * t)) *
-                exp(-((C * t - x)^2 / (4 * D * t))) for t in 1:t_max
+                exp(-((C * t - x)^2 / (4 * D * t))) for
```

Comment:
note to future selves: this (and the hillslope one) are allocated each step

Reply:
and this is allocated for each basin as well. (the gamma(t) should be able to be computed once, perhaps it can live in the HillslopeModel, since tmax lives there already, and this is specific to that model? But I do see that this channel one will be different for each basin depending on upstream distances. )

Reply:
Yes I think in future, we can compute this once for the graph,

Reply:
Note created